ledger-keyring-mv2.js

```diff
@@ -0,0 +1,195 @@
+const { BaseLedgerKeyring } = require('./base-ledger-keyring')
+
+const CONNECTION_EVENT = 'ledger-connection-change'
+
+class LedgerKeyringMv2 extends BaseLedgerKeyring {
+  init () {
+    this.iframeLoaded = false
+    this._setupIframe()
+
+    this.currentMessageId = 0
+    this.messageCallbacks = {}
+    this._setupListener()
+
+    return Promise.resolve()
+  }
+
+  destroy () {
+    window.removeEventListener('message', this._eventListener)
+
+    return Promise.resolve()
+  }
+
+  _setupIframe () {
+    this.iframe = document.createElement('iframe')
+    this.iframe.src = this.bridgeUrl
+    this.iframe.allow = `hid 'src'`
+    this.iframe.onload = async () => {
+      // If the ledger live preference was set before the iframe is loaded,
+      // set it after the iframe has loaded
+      this.iframeLoaded = true
+      if (this.delayedPromise) {
+        try {
+          const result = await this.updateTransportMethod(
+            this.delayedPromise.transportType,
+          )
+          this.delayedPromise.resolve(result)
+        } catch (e) {
+          this.delayedPromise.reject(e)
+        } finally {
+          delete this.delayedPromise
+        }
+      }
+    }
+    document.head.appendChild(this.iframe)
+  }
+
+  _setupListener () {
+    this._eventListener = ({ origin, data }) => {
+      if (origin !== this._getOrigin()) {
+        return false
+      }
+
+      if (data) {
+        if (this.messageCallbacks[data.messageId]) {
+          this.messageCallbacks[data.messageId](data)
+        } else if (data.action === CONNECTION_EVENT) {
+          this.isDeviceConnected = data.payload.connected
+        }
+      }
+
+      return undefined
+    }
+    window.addEventListener('message', this._eventListener)
+  }
+
+  attemptMakeApp () {
+    return new Promise((resolve, reject) => {
+      this._sendMessage(
+        {
+          action: 'ledger-make-app',
+        },
+        ({ success, error }) => {
+          if (success) {
+            resolve(true)
+          } else {
+            reject(error)
+          }
+        },
+      )
+    })
+  }
+
+  updateTransportMethod (transportType) {
+    return new Promise((resolve, reject) => {
+      // If the iframe isn't loaded yet, let's store the desired transportType value and
+      // optimistically return a successful promise
+      if (!this.iframeLoaded) {
```

Comment:
Does this function get called anywhere else but :33? Seems this block will never execute?

Reply:
It could if the iframe has not loaded yet, as `iframeLoaded` only changes to true when within the `iframe.onload` handle.

The code itself is a bit convoluted and slightly confusing, that's why I went for a rather conservative approach of changing as little as possible for this PR, as I didn't want to introduce any new bugs.